services/server/src/server/services/storageServices/AbstractDatabaseService.ts

```diff
@@ -0,0 +1,383 @@
+      });
+
+      // Add the onchain contract in contracts
+      const contractInsertResult = await Database.insertContract(
+        this.databasePool,
+        {
+          creationBytecodeHash: databaseColumns.bytecodeHashes.onchainCreation,
+          runtimeBytecodeHash: databaseColumns.bytecodeHashes.onchainRuntime,
+        }
+      );
+
+      // add the onchain contract in contract_deployments
+      await Database.insertContractDeployment(this.databasePool, {
+        chainId: match.chainId,
+        address: match.address,
+        transactionHash: match.creatorTxHash!,
+        contractId: contractInsertResult.rows[0].id,
+      });
+
+      // insert new recompiled contract
+      const compiledContractsInsertResult =
+        await Database.insertCompiledContract(this.databasePool, {
+          compiler: recompiledContract.compiledPath,
+          version: recompiledContract.compilerVersion,
+          language: databaseColumns.compiledContract.language!,
+          name: recompiledContract.name,
+          fullyQualifiedName:
+            databaseColumns.compiledContract.fullyQualifiedName!,
+          compilationArtifacts:
+            databaseColumns.compiledContract.compilationArtifacts!,
+          sources: recompiledContract.solidity,
+          compilerSettings: recompiledContract.metadata.settings,
+          creationCodeHash: databaseColumns.bytecodeHashes.recompiledCreation,
+          runtimeCodeHash: databaseColumns.bytecodeHashes.recompiledRuntime,
+          creationCodeArtifacts:
+            databaseColumns.compiledContract.creationCodeArtifacts!,
+          runtimeCodeArtifacts:
+            databaseColumns.compiledContract.runtimeCodeArtifacts!,
+        });
+
+      // insert new recompiled contract with newly added contract and compiledContract
+      const verifiedContractInsertResult =
+        await Database.insertVerifiedContract(this.databasePool, {
+          compilationId: compiledContractsInsertResult.rows[0].id,
+          contractId: contractInsertResult.rows[0].id,
+          creationTransformations:
+            databaseColumns.verifiedContract.creationTransformations,
+          creationTransformationValues:
+            databaseColumns.verifiedContract.creationTransformationValues || {},
+          runtimeTransformations:
+            databaseColumns.verifiedContract.runtimeTransformations,
+          runtimeTransformationValues:
+            databaseColumns.verifiedContract.runtimeTransformationValues || {},
+          runtimeMatch: databaseColumns.verifiedContract.runtimeMatch!,
+          creationMatch: databaseColumns.verifiedContract.creationMatch!,
+        });
+      return verifiedContractInsertResult.rows[0].id;
+    } catch (e) {
+      throw new Error(
+        `cannot insert verified_contract address=${match.address} chainId=${match.chainId}\n${e}`
+      );
+    }
+  }
+
+  async updateExistingVerifiedContract(
+    existingVerifiedContractResult: any,
```

Comment:
No, because Tables.VerifiedContract has column names with a camel case (to be ok for the linter), while the existingVerifiedContractResult is an array of objects with the properties with the same name as the table columns (e.g. runtimeTransformations vs runtime_transformations)

Reply:
I'll try to convert it into snake case, if it's not too much work

Reply:
Done